lib/string-checker.js

```diff
@@ -130,7 +130,12 @@ StringChecker.prototype = {
 
         this._configuredRules.forEach(function(rule) {
             errors.setCurrentRule(rule.getOptionName());
-            rule.check(file, errors);
+
+            try {
+                rule.check(file, errors);
+            } catch (e) {
+                errors.add(e, 1, 0);
```

Comment:
I don't see how we need to change API with custom Error (stringify-able) objects passed instead of plain string.

Atm we have one big Errors object with a list of messages and their locations (and some other flags).
I think we can drastically simplify that by making several Error types that will contain the same for code style errors (message, location, rule, fixed flag), and something else for other.

I'm not insist but just an idea.

Anyway, I don't now see how to make it work without hardcoding here output.